geoh5py/shared/utils.py

```diff
@@ -382,32 +382,34 @@ def dict_mapper(
     return val
 
 
-def box_intersect(
-    extent_a: np.ndarray, extent_b: np.ndarray, return_array=False
-) -> bool | np.ndarray:
+def box_intersect(extent_a: np.ndarray, extent_b: np.ndarray) -> bool:
     """
     Compute the intersection of two axis-aligned bounding extents defined by their
-    bottom south-west corner and top north-east corner.
+    arrays of minimum and maximum bounds in N-D space.
 
-    :param extent_a: First extent
-    :param extent_b: Second extent
-    :param return_array: Return the intersection extent
+    :param extent_a: First extent or shape (2, N)
+    :param extent_b: Second extent or shape (2, N)
 
-    :return: True if the box extents intersect,
-        or intersection extent coordinates if `return_array` is True.
+    :return: Logic if the box extents intersect along all dimensions.
     """
-    min_ext = []
-    max_ext = []
-    for comp_a, comp_b in zip(extent_a.T, extent_b.T):
-        min_ext.append(max(min(comp_a[0], comp_a[1]), min(comp_b[0], comp_b[1])))
-        max_ext.append(min(max(comp_a[0], comp_a[1]), max(comp_b[0], comp_b[1])))
+    for extent in [extent_a, extent_b]:
+        if not isinstance(extent, np.ndarray) or extent.ndim != 2:
+            raise TypeError("Input extents must be 2D numpy.ndarrays.")
+
+        if extent.shape[0] != 2 or not np.all(extent[0, :] <= extent[1, :]):
+            raise ValueError(
+                "Extents must be of shape (2, N) containing the minimum and maximum "
+                "bounds in nd-space on the first and second row respectively."
+            )
 
-    overlap = np.array([min_ext, max_ext])
+    for comp_a, comp_b in zip(extent_a.T, extent_b.T):
+        min_ext = max(min(comp_a[0], comp_a[1]), min(comp_b[0], comp_b[1]))
+        max_ext = min(max(comp_a[0], comp_a[1]), max(comp_b[0], comp_b[1]))
 
-    if return_array:
-        return overlap
+        if (max_ext - min_ext) < 0:
+            return False
 
-    return not (np.diff(overlap, axis=0) < 0).any()
+    return True
```

Comment:
**suggestion (non-blocking):** no need to resort to zip and translate. Can also simplify min/max  knowing that points are sorted
```py
    for i in range(min(extent.shape[0], extent.shape[1])):
        min_ext = max(comp_a[0][i], comp_b[0][i])
        max_ext = min(comp_a[1][i], comp_b[1][i])
        if min_ext > max_ext:
            return False
    return True
```

Reply:
I ll go half way. I think the zip is more elegant but granted that yours is more efficient.
Pushing.